core/src/main/java/bio/terra/pearl/core/service/notification/email/SendgridClient.java

```diff
@@ -71,16 +73,18 @@ public List<SendgridEvent> getEvents(Instant startDate, Instant endDate, int que
     return events;
   }
 
-  public Mail buildEmail(NotificationContextInfo contextInfo, String toAddress, String fromAddress,
+  public Mail buildEmail(NotificationContextInfo contextInfo, String toAddress, String fromAddress, String fromName,
                          StringSubstitutor stringSubstitutor) {
     Email from = new Email(fromAddress);
     Email to = new Email(toAddress);
 
-    String fromName = "Juniper";
-    if (contextInfo.portal() != null) {
-      // Set the 'from' name on the email to the portal name
-      from.setName(contextInfo.portal().getName());
+    if (fromName == null) {
+      fromName = "Juniper";
+    }
+    if (!deploymentZone.equalsIgnoreCase("prod")) {
```

Comment:
Not sure why both the environment and deployment zone are useful. It's a nit, but just curious since it looks kind of odd in the email header.

Reply:
at least for me, I get most of my emails from sandboxes, and so I want to know if it's local/prod/demo so I can tune it in/out appropriately, or withdraw the participant